sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -200,6 +219,30 @@ export class Receiver {
     return result;
   }
 
+  private getCheckpoint(): number | undefined {
+    if (this._streamingReceiver || this._batchingReceiver) {
+      let lastBatchingReceiverSequenceNum: number = -1;
+      let lastStreamingReceiverSequenceNum: number = -1;
+      if (this._batchingReceiver) {
+        lastBatchingReceiverSequenceNum = this._batchingReceiver.checkpoint;
+      }
+      if (this._streamingReceiver) {
+        lastStreamingReceiverSequenceNum = this._streamingReceiver.checkpoint;
+      }
+      return Math.max(lastStreamingReceiverSequenceNum, lastBatchingReceiverSequenceNum);
```

Comment:
Say the user called `receiveBatch()` and there were no events to return. We now would have a batchingReceiver with checkpoint as -1 and this function will return -1.
Meaning the next call from the user to `receiveBatch()`  ir `receive()` will start from the start of the stream instead of whatever the user provided as event position.